### PR TITLE
Resolved compilation and runtime errors

### PR DESCRIPTION
**Pull request by ES-UoL**
Code review for assignment 1

Development branch changes

Here are the changes that I have made via my latest push:

Directory:

The directory for the repository was slightly confusing as it seemed to have multiple instances of the code files in further subdirectories. I have restructured the directory to feature only the necessary set of code files by consolidating duplicate files and organizing them into a standard directory with one solution file to access all the code files to make it easier to navigate and work on the code base.

Testing class:
- The methods used to test the program are now qualified by the class instance name rather than the type.

Card class:
- Card constructor has been added to the Card class to be able to be called in the Pack constructor.

Pack class:
- Missing semicolon added on one of the lines to prevent a complialtion error.
- Every outcome of the shuffleCardPack now has a returned boolean value to match the method signature.
- Card pack list initialised as a static field to prevent compilation error.
- The _i variable initialised as a static field to prevent compilation error.
- The GetRange method has been modified to be used with the class instance rather than the class type to prevent a compilation error.
- The GetRange method has been modified to operate under the correct range and indexing to prevent a runtime error.
- Moved the first assignment of the flipFlop variable outside of the loop to prevent it from defaulting to false on each iteration.
_________________________________________________________________________________

Review
Here is my review, which is for the most recent version/upload that I could find of these code files. 

Documentation:

+ Good use of comments and well-formatted code, I was easily able to follow the code which helped me to be able to make the changes I have now made to the Development branch via my latest push.

+ Good implementation of the Testing class which allows the functionallity of each method to be tested separately

+ Good use of for loops which makes the code more efficient and easier to read.

Exception Handling:

+ Validation is present when the Card class properties are set to ensure that the cards cannot be created with values beyond what is expected.

Conclusion:

In conclusion, I would recommend reviewing the changes that I have made on my latest push to the Development branch to help you resolve the complilation and runtime errors. Additionally, the improvements that I would suggest you make would be to add exception handling to all possible inputs to ensure that everything is validated with sufficient feedback to the user. Adding a testing menu would be helpful to enable more specific testing of individual methods. An additional method could be created to display the cards so that you can evaluate how the shuffle algorithms and deal methods manipulate the cards in the deck. Overall, apart from the compilation and runtime errors, the code features what was asked for in the assignment brief. Great work! LGTM.

### CODE REVIEW

**Review by NathanVials-Student — APPROVED** [review]
Thanks Earl, I have looked through your changes and it all looks good and makes sense to me. Thank you for helping me out and sorting through some issues I had, you have been very helpful!
